pkg/pipeline/source/sdk/appwriter.go

```diff
@@ -354,38 +385,46 @@ func (w *AppWriter) pushBlankFrame(timestamp uint32) error {
 
 func (w *AppWriter) push(packets []*rtp.Packet, blankFrame bool) error {
 	for _, pkt := range packets {
-		// record timestamp diff
-		if w.rtpStep == 0 && !blankFrame && w.lastRTP != 0 && pkt.SequenceNumber == w.lastSN+1 {
-			w.rtpStep = pkt.Timestamp - w.lastRTP
-		}
-
-		// record SN and TS
-		w.lastSN = pkt.SequenceNumber
-		if w.lastRTP > wrapCheck && pkt.Timestamp < wrapCheck {
-			w.rtpWrap++
-		}
-		w.lastRTP = pkt.Timestamp
-
 		if !blankFrame {
 			// update sequence number
 			pkt.SequenceNumber += w.snOffset
+
+			// record max frame duration
+			if w.lastTS != 0 && pkt.SequenceNumber == w.lastSN+1 {
+				if frameDurationRTP := pkt.Timestamp - w.lastTS; frameDurationRTP > w.frameDurationRTP && frameDurationRTP < w.clockRate/60 {
+					w.frameDurationRTP = frameDurationRTP
+				}
+			}
+
 			w.translatePacket(pkt)
 		}
 
-		// will return io.EOF if EOS has been sent
+		// reset offsets if needed
+		if w.lastTS != 0 && pkt.SequenceNumber-w.lastSN > maxDropout && w.lastSN-pkt.SequenceNumber > maxDropout {
```

Comment:
Should the last `&&` be a `||` here? If not, can the test ever be true?

Reply:
these are uint32, so the && is checking that the absolute value of the difference is > maxDropout